### DISCUSSION TITLE
Spurious points in surfaces.vertices

### DISCUSSION THREAD

**Discussion by nlosacco** [gempy-project/gempy#585] [Q&A]
Hi all,

always stuck with trying to create a solid model, starting from creating a NURBS from the vertices of the surfaces. When I try and plot the vertices I see that they are not arranged in a regular grid,
rather some spurious points seem to be randomly generated. Is this an expected behaviour?
Please see the attached 3D plot.
Thanks,
N
![gempy_vertices](https://user-images.githubusercontent.com/31536633/84327775-85f7e200-ab80-11ea-933a-2db83fcf5b85.png)

**Comment by javoha** [comment]
Hi @nlosacco ,
if I understand correctly your are plotting the vertices of the resulting surface mesh (which are not NURBS just to be clear). These surface meshes are extracted from the scalar field using the marching cube method - therefore this is a "expected" result that does not align with the underlying regular grid. Using the gempy plotting functions should give you full surfaces and if you extract not just the vertices but also the simplices you should also be able to plot the complete surfaces mesh using other tools. If you can specify what you want to do with the result or what exactly you neeed, we can maybe provide more help.

Hope that helps and best wishes!

**Comment by nlosacco** [comment]
Hi @javoha thanks for your quick reply.
Yes, I skipped some steps in my question: the final aim is creating solid bodies that could be manipulated and meshed with external programs.
For this purpose I thought it might be good to try and create NURBS surfaces to start with, hence I would need data on a regular grid that can be processed by nurbs-python, for instance.
Clearly the spurious datapoints bias invalidate the process I had in mind and I need to try a different strategy.

**Comment by MohammadCGRE** [comment]
Dear @nlosacco 

if you are seeking for Solid object, you need NURBS-Volume. In the gempyExplicit, we defined the NURBS Volume (a solid object)..gempyExplicit is now in the experimental version..it will be acceptable for the public soon..but if you like, probably you can use it (i can make it acceptable for you..right now it is not jointed to gempy)...Can you explain to me what exactly you want?

**Comment by nlosacco** [comment]
Dear @MohammadCGRE ,

thanks for your advice. As usual (e.g. see #250,  #406), my aim is creating a Finite Element model for geotechnical applications in structurally complex formations (e.g. slope stability, tunnelling, etc.) to be analysed with various FE softwares, either open source or commercial.
I know one can export a simple cubic mesh into MOOSE  or generate a tetrahedral mesh with tetgen and so forth, but I would like to have a solid model that I can manipulate further and eventually mesh with external programs.
For this purpose I am struggling to export and eventually transform into NURBS the surfaces created by GemPy, that might be used, in turn, to create a volume in an external program and then generate the mesh.
Clearly, being able to generate a solid NURBS directly from GemPy, as I understand the NURBS-Volume function would do, that might be exported, for instance, as an IGES (.iges), STEP (.stp), ACIS (.sat) file, Rhino (.3dm) file, would be even better and solve the problem straightaway!
Thanks and congratulations for your great job.
